compiler/src/dmd/escape.d

```diff
@@ -1799,7 +1799,7 @@ void escapeByValue(Expression e, EscapeByResults* er, bool live = false, bool re
                         if (tf.isreturn)
                             stc |= STC.return_;
                         if (tf.isreturnscope)
-                            stc |= STC.returnScope;
+                            stc |= STC.returnScope | STC.scope_;
```

Comment:
It's nuts that we have to do this though...

Reply:
I agree, I intend to improve it in future PRs, this is just the quickest path to fix the regression. I've been looking already into fixing `returnScope` inference without `scope`, but it breaks Phobos somehow.